### PR TITLE
Fix some inconsistent online tests

### PR DESCRIPTION
**Pull request by Cadair**
This also adds a much better error message when we can't connect to any HELIO servers.

TODO:

- [x] A test for the helio error

### CODE REVIEW

File: sunpy/net/dataretriever/sources/rhessi.py [hunk 1]
```diff
@@ -62,8 +62,8 @@ def get_observing_summary_filename(self, time_range):
         --------
         >>> from sunpy.net.dataretriever.sources.rhessi import RHESSIClient
         >>> RHESSIClient().get_observing_summary_filename(('2011/04/04', '2011/04/05'))   # doctest: +REMOTE_DATA
-        ['https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits',
-        'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110405_031.fits']
+        ['https://.../hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits',
+        'https://.../hessidata/metadata/catalog/hsi_obssumm_20110405_031.fits']
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/rhessi.py]: I thought of that but then I was like, it makes the example opaque.

Comment (Cadair) [reply sunpy/net/dataretriever/sources/rhessi.py]: Yeah, it's not great but it's better than skipping it.

File: sunpy/net/tests/test_helio.py [hunk 2]
```diff
@@ -172,7 +171,8 @@ def test_wsdl_retriever_no_content(mock_endpoint_parser):
     """
     No links found? Return None
     """
-    assert wsdl_retriever() is None
+    with pytest.raises(ValueError):
+        assert wsdl_retriever() is None
```
Comment (dpshelio) [inline sunpy/net/tests/test_helio.py]: Shouldn't this be a simple call to the function without the `assert`

File: sunpy/net/tests/test_helio.py [hunk 3]
```diff
@@ -172,7 +171,8 @@ def test_wsdl_retriever_no_content(mock_endpoint_parser):
     """
     No links found? Return None
```
Comment (dpshelio) [inline sunpy/net/tests/test_helio.py]: This comment should change too?

**Comment by Cadair** [comment]
@dpshelio updated.

**Comment by Cadair** [comment]
alright @dpshelio I think I have actually done it now :)

**Pull request merged**